pkg/app/app.go

```diff
@@ -59,15 +63,19 @@ type App struct {
 	pendingStatusUpdate   bool
 	flushAllStatusUpdates bool
 	metadata              *deploy.Meta
+
+	coreClient kubernetes.Interface
```

Comment:
We shouldn't need the coreClient in our App, either one of the factories ought to handle interactions with the coreclient, or our compInfo struct. But we want to avoid passing the coreClient around to every newApp instance.